
syntax = "proto3";

package protobuf;

message Batch {

    message Transform {
        string inputType = 1; // Caffe/Protobuf/Gzip
        string inputDataType = 2;
        string outputType = 3; // Protobuf/Gzip
        string outputDataType = 4;
        string network = 5;
        bool activate_bias_and_out_act = 6;
    }

    message Simulate {

        message Experiment {
            string architecture = 1;
            string task = 2;

            // For grid accelerators
            uint32 n_columns = 3;
            uint32 n_rows = 4;

            // For pragmatic
            uint32 bits_first_stage = 5;

            // For tactical
            uint32 lookahead_h = 6;
            uint32 lookaside_d = 7;
            string search_shape = 8;
        }

        string inputType = 1;
        string inputDataType = 2;
        string network = 3;
        bool activate_bias_and_out_act = 4;
        repeated Experiment experiment = 5;
    }

    repeated Transform transform = 1;
    repeated Simulate simulate = 2;

}